girder/models/file.py

```diff
@@ -182,14 +183,26 @@ def createLinkFile(self, name, parent, parentType, url, creator, size=None, mime
         :type size: int
         :param mimeType: The mimeType of the file. (optional)
         :type mimeType: str
+        :param reuseExisting: If a file with the same name already exists in
+            this location, return it rather than creating a new file.
+        :type reuseExisting: bool
         """
+
         if parentType == 'folder':
             # Create a new item with the name of the file.
             item = self.model('item').createItem(
-                name=name, creator=creator, folder=parent)
+                name=name, creator=creator, folder=parent, reuseExisting=reuseExisting)
         elif parentType == 'item':
             item = parent
 
+        if reuseExisting:
+            existing = self.findOne({
+                'itemId': item['_id'],
+                'name': name
+            })
+            if existing:
+                return existing
```

Comment:
I'm not sure. How would it work for regular files? Would you update the content?